elasticsearch/bulk/bulk.go

```diff
@@ -122,22 +131,29 @@ func (b *Bulk) AddActions(
 		b.flushLock.Unlock()
 		return
 	}
-	for _, action := range actions {
+	for i, action := range actions {
 		indexName := b.getIndexName(collectionName, action.IndexName)
+		actions[i].IndexName = indexName
 		value := getEsActionJSON(
 			action.ID,
 			action.Type,
-			indexName,
+			actions[i].IndexName,
```

Comment:
can we use "indexName" here to access index name directly instead of address resolution calculation?

Reply:
Yeap, i can revert this part. But, i thought we need to manage index from action. When i looked ESActionDocument creation, there is not any indexName setting field.

```

func NewDeleteAction(key []byte, routing *string) ESActionDocument {
	return ESActionDocument{
		ID:      key,
		Routing: routing,
		Type:    Delete,
	}
}

func NewIndexAction(key []byte, source []byte, routing *string) ESActionDocument {
	return ESActionDocument{
		ID:      key,
		Routing: routing,
		Source:  source,
		Type:    Index,
	}
}
```

We are managing actions[i].IndexName field as follows

```
indexName := b.getIndexName(collectionName, action.IndexName)
actions[i].IndexName = indexName
```

I think, it can stay 🤔 wdyt ?

Reply:
You are right we need to update NewDeleteAction and NewIndexAction as well, it should take parameter as indexName *string

Reply:
I understand the usage right now in bytes you don't know the index